### PR TITLE
DM-40443: Remove interfaces deprecated on DM-40441

### PR DESCRIPTION
**Pull request by timj**
Paired with lsst/pipe_base#434

## Checklist

- [x] ran Jenkins
- [x] added a release note for user-visible changes to `doc/changes`

### CODE REVIEW

File: python/lsst/ctrl/mpexec/dotTools.py [hunk 1]
```diff
@@ -282,10 +281,8 @@ def expand_dimensions(connection: connectionTypes.BaseConnection) -> list[str]:
     allDatasets: set[str | tuple[str, str]] = set()
     if isinstance(pipeline, Pipeline):
         # TODO: DM-40639 will rewrite this code and finish off the deprecation
-        # of toExpandedPipeline.
-        with warnings.catch_warnings():
-            warnings.simplefilter("ignore", category=FutureWarning)
-            pipeline = pipeline.toExpandedPipeline()
+        # of toExpandedPipeline but for now use the compatibility API.
+        pipeline = pipeline.to_graph()._iter_task_defs()
```
Comment (timj) [inline python/lsst/ctrl/mpexec/dotTools.py:285]: @TallJimbo I assume this is okay for now (I got it from the old definition of `toExpandedPipeline`).

File: python/lsst/ctrl/mpexec/mpGraphExecutor.py [hunk 2]
```diff
@@ -169,7 +169,7 @@ def _executeJob(
         quantum = pickle.loads(quantum_pickle)
         report: QuantumReport | None = None
         try:
-            _, report = quantumExecutor.execute(task_node, quantum)
+            quantum, report = quantumExecutor.execute(task_node, quantum)
```
Comment (timj) [inline python/lsst/ctrl/mpexec/mpGraphExecutor.py:172]: @TallJimbo  Is there a reason this was `_` before? I can put it back (I was experimenting with adding a cast around that `pickle.loads` and forgot to back this line out.

**Review by timj — COMMENTED** [review]
Looks good.

File: python/lsst/ctrl/mpexec/dotTools.py (hunk 1 above)
Comment (TallJimbo) [reply python/lsst/ctrl/mpexec/dotTools.py:285]: Yes, `_iter_task_defs` will stick around until `TaskDef` goes away.

File: python/lsst/ctrl/mpexec/mpGraphExecutor.py (hunk 2 above)
Comment (TallJimbo) [reply python/lsst/ctrl/mpexec/mpGraphExecutor.py:172]: I think it wasn't used until the `QuantumReport` stuff below was added, which was after I created this branch.

**Review by TallJimbo — APPROVED** [review]
Hitting approve, since you created the PR and I answered some questions.

**Pull request merged**